src/rascal/representations/calculator_spherical_invariants.hh

```diff
@@ -308,6 +311,84 @@ namespace rascal {
         Invariants & soap_vector, ExpansionCoeff & expansions_coefficients,
         std::shared_ptr<StructureManager> manager);
 
+    /**
+     * Update the gradients \grad_i p^{k} to include normalization, N_i,
+     * resulting in \grad_i \tilde{p}^{k}.
+     * We have:
+     * \grad_i \tilde{p}^{k} = \grad_i p^{k} / N_k
+             - \tilde{p}^{k} [\tilde{p}^{k} \cdot \grad_i p^{k} / N_k],
+     * where $\cdot$ is a dot product between vectors.
+     * Note that this expects the soap vectors to be normalized already, and
+     * the norm stored separately
+     */
+    template <class StructureManager, class SpectrumNorm>
+    void normalize_gradients(
```

Comment:
This naming isn't great -- it implies that the gradients are being divided by their norms, when the actual operation (written above) is more complicated. I'd prefer something like `update_gradients_for_normalization`.